sdk/core/azure-core/src/test/java/com/azure/core/implementation/models/jsonflatten/SampleResource.java

```diff
@@ -37,5 +42,52 @@ public String getNamePropertiesName() {
     public String getRegistrationTtl() {
         return registrationTtl;
     }
+
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) {
+        jsonWriter.writeStartObject();
+
+        if (namePropertiesName == null && registrationTtl == null) {
+            return jsonWriter.writeEndObject().flush();
+        }
+
+        jsonWriter.writeFieldName("properties").writeStartObject();
+
+        JsonUtils.writeNonNullStringField(jsonWriter, "name", namePropertiesName);
+        JsonUtils.writeNonNullStringField(jsonWriter, "registrationTtl", registrationTtl);
+
+        return jsonWriter.writeEndObject().writeEndObject().flush();
+    }
+
+    public static SampleResource fromJson(JsonReader jsonReader) {
+        return JsonUtils.readObject(jsonReader, (reader, token) -> {
+            String namePropertiesName = null;
+            String registrationTtl = null;
+
+            while (reader.nextToken() != JsonToken.END_OBJECT) {
+                String fieldName = reader.getFieldName();
+                token = reader.nextToken();
+
+                if ("properties".equals(fieldName) && token == JsonToken.START_OBJECT) {
+                    while (reader.nextToken() != JsonToken.END_OBJECT) {
+                        fieldName = reader.getFieldName();
+                        reader.nextToken();
+
+                        if ("name".equals(fieldName)) {
+                            namePropertiesName = reader.getStringValue();
+                        } else if ("registrationTtl".equals(fieldName)) {
+                            registrationTtl = reader.getStringValue();
+                        } else {
+                            reader.skipChildren();
```

Comment:
TODO: all unknown properties should skipChildren, either the current token is a simple value like boolean, null, number, or string and this is a no-op and progression in the next loop will skip it. Or the value is pointing to a sub-object or array which will need to perform recursive skipping.

There is a big question on how this will be handled when there is additional properties on the class.

Reply:
curious, are there valid cases like these where we have unknown property without additional properties. Would be great to write a warning there and eventually throw and try running integration tests to understand the scale of it.